name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/heatmap/ColorSchema.java

```diff
@@ -24,46 +26,129 @@ private ColorSchema(String label)
         this.label = label;
     }
 
-    @Override
-    public String toString()
+    DoubleFunction<Color> buildColorFunction(ResourceManager resourceManager)
     {
-        return label;
+        return switch (this)
+        {
+            case GREEN_YELLOW_RED -> performance -> {
+                // Normalize performance between -0.07 and +0.07 and map it to a
+                // hue-like scale between 0 (red) and 60 (yellow) or 120
+                // (green).
+                double p = normalizePerformance(performance);
+                float hue = (float) p * 120f;
+
+                return resourceManager.createColor(new RGB(hue, 0.9f, 1f));
+            };
+
+            case GREEN_WHITE_RED -> performance -> {
+                // Performance is normalized and interpolated between green
+                // (positive) and the background (neutral) or red (negative)
+                double p = Math.min(MAX_PERFORMANCE, Math.abs(performance)) / MAX_PERFORMANCE;
+
+                RGB color;
+
+                if (performance > 0f)
+                {
+                    color = Colors.interpolate(Colors.theme().defaultBackground().getRGB(),
+                                    Colors.HEATMAP_DARK_GREEN.getRGB(), (float) p);
+
+                    // Adjusted transition to yellow earlier
+                    if (p > 0.4)
+                    {
+                        // 0.4 to 1.0 to 0 to 1.0
+                        float yellowFactor = (float) ((p - 0.4) / 0.6);
+                        color = Colors.interpolate(color, Colors.YELLOW.getRGB(), yellowFactor);
+                    }
+                }
+                else
+                {
+                    // Using -p for correct mapping of negatives
+                    color = Colors.interpolate(Colors.theme().defaultBackground().getRGB(), Colors.RED.getRGB(),
+                                    (float) -p);
```

Comment:
By using ```-p``` all negative values will be white (= the color of the background) because the ```interpolate``` function clamps the value between 0 and 1 (and p is always between 0 and 1)